Manager changes for v7 on-demand store

tmeasday opened the pull request:
Part of https://github.com/storybookjs/storybook/pull/15871

Telescoping on https://github.com/storybookjs/storybook/pull/15999

## What I did

Changes in the manager to:

 - Fetch `stories.json` and use it to build the sidebar if it is v3 data
 - Deal with the new `STORY_PREPARED` event
 - Add a `useStoryPrepared()` hook for addons to use to show a loading state while they wait for the story to be prepared.

tmeasday committed:
Manager changes for v7 on-demand store

shilman commented on lib/api/src/modules/stories.ts:
```diff
@@ -327,9 +348,53 @@ export const init: ModuleFn = ({
+
+      // We can only do this if the stories.json is a proper storyIndex
+      if (storyIndex.v !== 3) {
+        return;
```
Log a warning here at least

tmeasday replied:
This happens if you are running a built story book with stories.json on the v6 store. Do we want to log a warning?

shilman replied:
I think we should log something, even if it's as simple as `Skipping story index with version v${storyIndex.v}`.

tmeasday replied:
OK

tmeasday committed:
Add warning if find old `stories.json`

https://github.com/storybookjs/storybook/pull/16004/files/b05020ce5c1ca9f12cc5c0f7bd4e2aa8d20d7102#r705834795

shilman commented:
Merged as part of https://github.com/storybookjs/storybook/pull/15871
